lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -176,6 +176,57 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     end
   end
 
+  @doc """
+
+  """
+  @spec fetch_identifier(
+          charlist(),
+          {:module, module()} | {:function | :type, name :: atom(), arity :: pos_integer()}
+        ) ::
+          {:ok, cell_id :: String.t(), line :: pos_integer()} | :error
+  def fetch_identifier(path, identifier)
+
+  def fetch_identifier(path, {:module, module}) do
+    with {:ok, {:raw_abstract_v1, annotations}} <- beam_lib_chunks(path, :abstract_code) do
+      {:attribute, anno, :module, ^module} =
+        Enum.find(annotations, &match?({:attribute, _, :module, _}, &1))
+
+      {:ok, cell_id_from_module(path), :erl_anno.line(anno)}
+    end
+  end
+
+  def fetch_identifier(path, {:function, name, arity}) do
+    with {:ok, {:debug_info_v1, _, {_, metadata, _}}} <- beam_lib_chunks(path, :debug_info) do
+      {_, :def, [line: line], _} =
+        Enum.find(metadata.definitions, &match?({{^name, ^arity}, :def, [line: _line], _}, &1))
+
+      {:ok, cell_id_from_module(path), line}
```

Comment:
`:line` may not be the only field in `meta`, so I would go with something like this:

```suggestion
    with {:ok, {:debug_info_v1, _, {_, metadata, _}}} <- beam_lib_chunks(path, :debug_info),
         {_pair, _kind, meta, _body} <- keyfind(metadata.definitions, {name, arity}),
         {:ok, line} <- Keyword.fetch(meta, :line) do
      {:ok, cell_id_from_module(path), line}
```

where:

```
    defp keyfind(list, key) do
      List.keyfind(list, key, 0) || :error
    end
```